vid parsing regex incorrect

nzaillian opened the pull request:
Hi, the current vid-parsing regex just uses the "w+" token and doesn't capture hyphens or underscores (which can be in youtube vids). For example, try the following (this is a real youtube video that I was trying to fetch info for in my app):

```
client = YouTubeIt::Client.new
video = client.video_by("http://www.youtube.com/watch?v=CBvFZV-jdHY")
```

Because there's a hyphen in that vid, all that gets parsed out currently is "CBvFZV" and, basically, everything fails. Please merge this when you can (just changed regex to match all alphanumeric strings including hyphens and underscores). I'm monkey-patching over the method right now in an initializer (I'm using the library in a Rails app) to get around this problem, which is pretty critical

nzaillian committed:
fix for vid-parsing regex (previously did not capture hyphen or
underscore, which are valid vid components)

kylejginavan commented:
Fix the test as well then I will get this merged!  Thanks for the contribution.

nzaillian commented:
Added an assertion with the url that had been failing to parse before. The test in question was commented out so I left it that way. The last test, which was the only one in the file that wasn't commented out ("test_get_all_videos") was failing because it seems the youtube user ("enchufetv") who's account it queries against has added an extra video(?), so I updated that too
